src/v2/view-builder/views/ov/NumberChallengePushView.js

```diff
@@ -0,0 +1,78 @@
+import { _, loc } from 'okta';
+import BaseForm from '../../internals/BaseForm';
+import polling from '../shared/polling';
+import { WARNING_TIMEOUT } from '../../utils/Constants';
+import ResendNumberChallengeView from './ResendNumberChallengeView';
+import NumberChallengePhoneView from './NumberChallengePhoneView';
+
+const Body = BaseForm.extend(Object.assign(
+  {
+    noButtonBar: true,
+
+    className: 'okta-verify-number-challenge',
+
+    events: {
+      'click a.resend-number-challenge': 'handleResendNumberChallenge'
+    },
+
+    handleResendNumberChallenge () {
+      this.options.appState.trigger('invokeAction', 'currentAuthenticator-resend');
+      // remove the warning
+      this.stopPoll();
+      // restart polling
+      this.startPoll();
+    },
+
+    title () {
+      return loc('oie.okta_verify.push.sent', 'login');
+    },
+
+    initialize () {
+      BaseForm.prototype.initialize.apply(this, arguments);
+      this.add(NumberChallengePhoneView);
+    },
+
+    triggerAfterError () {
+      BaseForm.prototype.triggerAfterError.apply(this, arguments);
+      this.stopPolling();
+      this.$el.find('.o-form-fieldset-container').empty();
+    },
+
+    postRender () {
+      this.startPoll();
+    },
+
+    startPoll () {
+      this.startPolling();
+      this.warningTimeout = setTimeout(_.bind(function () {
+        this.showWarning();
+      }, this), WARNING_TIMEOUT);
+    },
+
+    stopPoll () {
+      this.stopPolling();
+      this.clearWarning();
+    },
+
+    showWarning () {
+      this.clearWarning();
+      this.add(ResendNumberChallengeView, '.o-form-error-container');
```

Comment:
Why do we need to call `this.clearWarning` beforehand? It appears that we'll only call `showWarning` once after the view renders.

Reply:
you are right, we can remove it